src/pybind/mgr/ansible/ansible_runner_svc.py

```diff
@@ -0,0 +1,271 @@
+            status_value = ExecutionStatusCode.ERROR
+        else:
+            endpoint = "%s/%s" % (PLAYBOOK_EXEC_URL, self.play_uuid)
+            response = self.rest_client.http_get(endpoint)
+
+            if response:
+                the_status = json.loads(response.text)["msg"]
+                if the_status == 'successful':
+                    status_value = ExecutionStatusCode.SUCCESS
+                elif the_status == 'failed':
+                    status_value = ExecutionStatusCode.ERROR
+                else:
+                    status_value = ExecutionStatusCode.ON_GOING
+            else:
+                status_value = ExecutionStatusCode.ERROR
+
+        self.log.info("Requested playbook execution status is: %s", status_value)
+        return status_value
+
+    def get_result(self, event_filter=""):
+        """Get the data of the events filtered by a task pattern and
+        a event filter
+
+        @returns: the events that matches with the patterns provided
+        """
+
+        if not self.result_task_pattern or not self.play_uuid:
+            result_events = {}
+
+        response = self.rest_client.http_get(PLAYBOOK_EVENTS % self.play_uuid)
+
+        if not response:
+            result_events = {}
+        else:
+            events = json.loads(response.text)["data"]["events"]
+            result_events = {event:data for event,data in events.items()
+                            if "task" in data and
+                            re.match(self.result_task_pattern, data["task"])}
+            if event_filter:
+                result_events = {event:data for event,data in result_events.items()
+                                if re.match(event_filter, data['event'])}
+
+        self.log.info("Requested playbook result is: %s", json.dumps(result_events))
+        return result_events
+
+class Client(object):
+    """An utility object that allows to connect with the Ansible runner service
+    and execute easily playbooks
+    """
+
+    def __init__(self, server, port, user, password, certificate, logger):
+        """Get the Ansible runner service url and user credentials to be used in
+        all the requests.
+
+        @param ar_service_url: The URL of the Ansible runner service
+        @param user: A tuple with (user_name, password), the user authorized to
+                    connect with the Ansible Runner service
+        """
+        self.server = server
+        self.port = port
+        self.user = user
+        self.password = password
+        self.log = logger
+        self.auth = (self.user, self.password)
+
+        # Used to verify or not https server identity
+        if not certificate:
+            self.certificate = False
+        else:
+            self.certificate = certificate
+
+        # Once authenticated this token will be used in all the requests
+        self.token = ""
+
+        self.server_url = "https://{0}:{1}".format(self.server, self.port)
```

Comment:
This may not IPv6 safe. For simplicity reasons, I'd suggest to not assemble URLs by hand.

Reply:
Changed: Thx!
server, port settings are changed to only one setting: server_url